src/key.rs

```diff
@@ -411,17 +503,53 @@ impl PublicKey {
         }
     }
 
-    /// Adds a second key to this one, returning the sum. Returns an error if
-    /// the result would be the point at infinity, i.e. we are adding this point
-    /// to its own negation
+    /// Adds a second key to this one, returning the sum.
+    ///
+    /// # Errors
+    ///
+    /// If the result would be the point at infinity, i.e. adding this point to its own negation.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(feature="rand")] {
+    /// use secp256k1::Secp256k1;
+    /// use secp256k1::rand::{RngCore, thread_rng};
+    ///
+    /// let secp = Secp256k1::new();
+    /// let mut rng = thread_rng();
+    /// let (_, pk1) = secp.generate_keypair(&mut rng);
+    /// let (_, pk2) = secp.generate_keypair(&mut rng);
+    /// let sum = pk1.combine(&pk2).expect("sum");
```

Comment:
Maybe replace `.expect("sum")` with `unwrap` or `expect("It's improbable to fail for 2 random public keys")`